mikhail.kostyukov/task-5/pkg/handlers/handlers.go

```diff
@@ -0,0 +1,125 @@
+package handlers
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"sync"
+)
+
+const (
+	tagDecorated      = "decorated: "
+	stopDecorator     = "no decorator"
+	stopMultiplexer   = "no multiplexer"
+	msgCannotDecorate = "can't be decorated"
+	msgEmptyOutputs   = "empty outputs channels"
+)
+
+var (
+	ErrDecorationRefused = errors.New(msgCannotDecorate)
+	ErrEmptyOutputs      = errors.New(msgEmptyOutputs)
+)
+
+func PrefixDecoratorFunc(
+	ctx context.Context,
+	input chan string,
+	output chan string,
+) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case message, isOpen := <-input:
```

Comment:
```suggestion
		case message, ok := <-input:
```

и че вы все к этому `ok` прикопались... какой-то конкурс "кто сильнее извернется"